mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -102,33 +151,80 @@ class Completions(
   inline private def undoBacktick(label: String): String =
     label.stripPrefix("`").stripSuffix("`")
 
-  private def findSuffix(methodSymbol: Symbol): Option[String] =
-    if shouldAddSnippet && methodSymbol.is(Flags.Method) then
-      lazy val extensionParam = methodSymbol.extensionParam
+  private def getParams(symbol: Symbol) =
+    lazy val extensionParam = symbol.extensionParam
+    if symbol.is(Flags.Extension) then
+      symbol.paramSymss.filter(
+        !_.contains(extensionParam)
+      )
+    else symbol.paramSymss
+
+  private def findSuffix(symbol: Symbol): Option[String] =
+
+    val bracketSuffix =
+      if shouldAddSnippet && noSquareBracketExists
+        && (isTypePosition || isNewPosition || isInstantiationOrMethodCallPos)
+        && (symbol.info.typeParams.nonEmpty
+          || (symbol.isAllOf(
+            Flags.JavaModule
+          ) && symbol.companionClass.typeParams.nonEmpty))
+      then "[$0]"
+      else ""
+
+    val bracesSuffix =
+      if shouldAddSnippet && symbol.is(
+          Flags.Method
+        ) && !isTypePosition
+      then
+        val paramss = getParams(symbol)
+        paramss match
+          case Nil => ""
+          case List(Nil) => "()"
+          case _ if config.isCompletionSnippetsEnabled =>
+            val onlyParameterless = paramss.forall(_.isEmpty)
+            lazy val onlyImplicitOrTypeParams = paramss.forall(
+              _.exists { sym =>
+                sym.isType || sym.is(Implicit) || sym.is(Given)
+              }
+            )
+            if onlyParameterless then "()" * paramss.length
+            else if onlyImplicitOrTypeParams then ""
```

Comment:
what if we have `foo()(implicit x: Int) = ???`?
I guess both `onlyParameterless` and `onlyImplicitOrTypeParams` will be `false`, but `onlyImplicitOrTypeParams` should be `true`?

Reply:
Thank you so much for taking the time to look into this. 

Actually, this was the preexisting parenthesis suffix code. I only changed the return type from Option[String] to String. I did not touch the logic otherwise, just in case this would break something.

Reply:
okay, but it should be fixed anyway. You can fix it in this PR or open an issue to track the issue :)

Reply:
I don't think so, because

- `foo()(implicit x: int)` will have `paramss = List(Nil, List(implicit x))`.
-  `Nil.exists(pred)` will be false for any pred
- therefore, `paramss.forall(...)` will be false

no?

Reply:
Yeah, let's check how it works.
even if it works weirdly, you don't need to fix that in this PR :+1:

Reply:
```
checkSnippet(
    "empty-params-with-implicit".only,
    s"""|object Main {
        |  def doSomething()(implicit x: Int) = x
        |  val bar = doSomethi@@
        |}
        |""".stripMargin,
      "doSomething($0)",
  )
```
passing 👍